File: android/core/src/main/java/uk/co/amlcurran/social/TimeOfDay.java
package uk.co.amlcurran.social;

import java.util.concurrent.TimeUnit;

import javax.annotation.Nonnull;

public class TimeOfDay {

    private final long millis;

    @Nonnull
    public static TimeOfDay fromHours(int hours) {
        return new TimeOfDay(TimeUnit.HOURS.toMillis(hours));
    }

    @Nonnull
    public static TimeOfDay fromHoursAndMinute(int hours, int minutes) {
        return new TimeOfDay(TimeUnit.HOURS.toMillis(hours) + TimeUnit.MINUTES.toMillis(minutes));
    }

    private TimeOfDay(long millis) {
        this.millis = millis;
    }

    public long hoursInDay() {
        return TimeUnit.MILLISECONDS.toHours(millis);
    }

    public long minutesInDay() {
        return TimeUnit.MILLISECONDS.toMinutes(millis);
    }
}

File: android/shared/src/commonMain/kotlin/uk/co/amlcurran/social/TimeOfDay.kt
package uk.co.amlcurran.social

import java.util.concurrent.TimeUnit

class TimeOfDay private constructor(private val millis: Long) {

    fun hoursInDay(): Long {
        return TimeUnit.MILLISECONDS.toHours(millis)
    }

    fun minutesInDay(): Long {
        return TimeUnit.MILLISECONDS.toMinutes(millis)
    }

    companion object {

        fun fromHours(hours: Int): TimeOfDay {
            return TimeOfDay(TimeUnit.HOURS.toMillis(hours.toLong()))
        }

    }
}
